File: program22.js
// Write a javascript program to check whether a number is perfect or not.

function checkPerfect(iNo)
{
    let iCnt = 0;
    let iSum = 0;

    for(iCnt = 1; iCnt <= (iNo / 2); iCnt++)
    {
        if(iNo % iCnt == 0)
        {
            iSum = iSum + iCnt;
        }

        if(iSum > iNo)
        {
            break;
        }
    }

    if(iNo == iSum)
    {
        return true;
    }
    else
    {
        return false;
    }
}

let iValue = 28;
let bRet = false;

bRet = checkPerfect(iValue);

if(bRet == true)
{
    console.log("Number is Perfect");
}
else
{
    console.log("Number is not Perfect");
}

File: Advanced LB Questions/program22.java
// Accept a number from user and check whether number is perfect or not
// A number is perfect if the sum of its divisors is equal to number itself.

import java.util.Scanner;

class program22
{
    public static boolean isPerfect(int iNo)
    {
        int iSum = 0;
        for(int i = 1; i <= (iNo / 2); i++)
        {
            if(iNo % i == 0)
            {
                iSum = iSum + i;
            }
        }

        return iSum == iNo;
    }

    public static void main(String[] args) 
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter a number : ");
        int iNo = sobj.nextInt();

        if(isPerfect(iNo))
            System.out.println("Number is perfect");
        else
            System.out.println("Number is not perfect");
    }
}